### ISSUE TITLE
Spaces at End of Search Terms Are Truncated

### ISSUE THREAD

**Issue by NoSuck** [qutebrowser/qutebrowser#874]
Currently, there doesn't appear to be a way to search an open page (/) for something with a space at the end, because trailing spaces are automatically deleted.

I love you, qutebrowser.  Marry me.

**Comment by The-Compiler** [comment]
I could swear I had an issue open for this already or even already fixed it, but apparently not. :wink:

**Comment by NoSuck** [comment]
I thought there was an issue for this, too!  Maybe it's #248 or #288 we're thinking of?

**Comment by The-Compiler** [comment]
So doing `/foo ` internally calls `:search -- /foo ` which will use `split.simple_split()` (because it has `maxsplit=0` set). That will strip trailing whitespace... I'll have to think whether I should change this behavior generally and what other commands that'd impact.

Impacted commands:

```
:quickmark-del
:bookmark-del
:openurl
:spawn
:quickmark-load
:bookmark-load
:search
:jseval
:debug-trace
:debug-pyeval
:set-cmd-text-command
```

**Comment by Carpetsmoker** [comment]
The best way to fix this is probably to add some quoting syntax to the command parser? For example:

    :search -- 'foo '

Simply maintaining the spaces seems undesirable to me, since >95% of the time you *want* trailing spaces removed from commands.

**Comment by lahwaacz** [comment]
To not reiterate this over and over, see #2017 for problems with quoting rules in commands and suggested solutions.

**Comment by Hi-Angel** [comment]
Just a ⁺¹ — I wanted to highlight defined Python functions in a github code, but turned out typing `def ` with the space doesn't work. I feel an irony there — Qutebrowser is written in Python.

**Comment by The-Compiler** [comment]
This was fixed by @jgkamat in #3796 by internally calling the search code directly (instead of relying on the command parser).
